src/features/ProfilePage/components/PublicInfo/index.tsx

```diff
@@ -100,18 +135,21 @@ const PublicInfo = ({ isMobile = false }: Props) => {
       <Form>
         <Text>{t('public.mentor.mandatoryNotice')}</Text>
         <Columns
-          isBirthYearInvalid={isBirthYearInvalid}
-          isDisplayNameTooShort={isDisplayNameTooShort}
+          birthYearError={getBirthYearError()}
```

Comment:
Would it make sense to just create a variable
`const birthYearError = getBirthYearError()`

and use that in the `isSavingDisabled` variable and in the component prop? Now we create one extra function for each.

Even better would be to 

```typescript
const error = getError()

const getError = (): Error | null => {
  if(it_is_this_error){
   return 'this_error'
}
...
return null
}


const errorMap = Record<Error, Message> = {
  this_error: this.error.localization,
  that_error: that.error.localization
}

const isSavingDisabled = Boolean(error) || !isDIrty || isLoading

return (
<Component error={error ? errorMap[error] : null} />
)
```

Now we have a lot of extra props.